lib/ClusterShell/Task.py

```diff
@@ -510,6 +510,8 @@ def set_info(self, info_key, value):
           - "command_timeout": Time in seconds to wait for a command to
             complete before aborting (default: 0, which means
             unlimited).
+          - "local_workername": In tree mode, the name of the local worker
```

Comment:
The current documented way to change the local worker is by changing Defaults. Since Defaults don't propagate in tree mode but a Task's Info does, that is (currently) the only way to convey the desired localworker to the gateway. I would welcome a solution that uses the same method for non-tree and tree modes; but we can't change the current expectation around using Defaults in non-tree mode, and I'm not sure that sending Defaults over tree mode makes sense (and is more code than this current patch).

I'm willing to implement this differently if you think another path is superior.

Thanks for reviewing.

Reply:
Would it be better if this were called `tree_local_workername` to make it more obvious that it only affects tree mode?